volatility3/framework/plugins/linux/pstree.py

```diff
@@ -12,44 +11,74 @@ class PsTree(pslist.PsList):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self._processes = {}
+        self._tasks = {}
         self._levels = {}
         self._children = {}
 
-    def find_level(self, pid):
-        """Finds how deep the pid is in the processes list."""
-        seen = set([])
-        seen.add(pid)
+    def find_level(self, pid: int) -> None:
+        """Finds how deep the PID is in the tasks hierarchy.
+
+        Args:
+            pid: PID to find the level in the hierachy
+        """
+        seen = set([pid])
         level = 0
-        proc = self._processes.get(pid, None)
-        while proc is not None and proc.parent != 0 and proc.parent.pid not in seen:
-            ppid = int(proc.parent.pid)
+        proc = self._tasks.get(pid)
+        while proc and proc.parent and proc.parent.pid not in seen:
+            if self.task_is_thread_group_leader(proc):
+                parent_pid = proc.parent.pid
+            else:
+                parent_pid = proc.tgid
 
-            child_list = self._children.get(ppid, set([]))
+            child_list = self._children.setdefault(parent_pid, set())
             child_list.add(proc.pid)
-            self._children[ppid] = child_list
-            proc = self._processes.get(ppid, None)
+
+            proc = self._tasks.get(parent_pid)
```

Comment:
Please provide a default (as we did previously) when calling get.  We can't be certain the `parent_pid` will definitely be in the list, so there should be a fallback that's handled.  If it's not this will throw a random `KeyError` to a user who won't know why...

`self._tasks`, unlike `self._children` has not has `setdefault` called on it, and if you *can* be guaranteed of the default value, then we don't need to use get we can just do index/key accesses.

Reply:
> Please provide a default (as we did previously) when calling get. We can't be certain the `parent_pid` will definitely be in the list, so there should be a fallback that's handled. If it's not this will throw a random `KeyError` to a user who won't know why...
Agree, but that's exactly what get() does. Its default `default` is **None**, so `get(something, None)` looks redundant to me.
```python
>>> help({})
...
 |  get(self, key, default=None, /)
 |      Return the value for key if key is in the dictionary, else default.
```